core/input/gesturehandler.js

```diff
@@ -234,7 +234,8 @@ export default class GestureHandler {
                     }
                 } else if (!this._isTwoTouchTimeoutRunning()) {
                     // We can't determine the gesture right now, let's
-                    // wait and see if more events are on their way
+                    // wait and see if more events are on their way.
+                    // If not, we'll have to decide which gesture it is.
                     this._startTwoTouchTimeout();
```

Comment:
This breaks other two finger gestures as the two touch timeout blindly assumes that only a pinch or drag is left to choose from. It might need to be fixed so it also just excludes things, leaving it still uncertain for any other gestures that have yet to be ruled out.

I'm not sure if this timeout was ever really used, or at least very little, given the existing large movement threshold. We might need to reexamine it in detail.

Reply:
The lack of distance threshold also means that the timeout handler has much more noisy data to deal with when trying to identify a pinch from a drag. So it might be much less robust there as well.